include/mata/nfa.hh

```diff
@@ -263,29 +263,296 @@ Mata::Parser::ParsedSection serialize(
+    explicit Move(Symbol symbolOnTransition) : symbol(symbolOnTransition), targets() {}
     Move(Symbol symbolOnTransition, State states_to) :
-            symbol(symbolOnTransition), states_to{states_to} {}
+            symbol(symbolOnTransition), targets{states_to} {}
     Move(Symbol symbolOnTransition, const StateSet& states_to) :
-            symbol(symbolOnTransition), states_to(states_to) {}
+            symbol(symbolOnTransition), targets(states_to) {}
 
     inline bool operator<(const Move& rhs) const { return symbol < rhs.symbol; }
     inline bool operator<=(const Move& rhs) const { return symbol <= rhs.symbol; }
-    inline bool operator>(const Move& rhs) const { return symbol > rhs.symbol; }
-    inline bool operator>=(const Move& rhs) const { return symbol >= rhs.symbol; }
     inline bool operator==(const Move& rhs) const { return symbol == rhs.symbol; }
     inline bool operator!=(const Move& rhs) const { return symbol != rhs.symbol; }
+    inline bool operator>(const Move& rhs) const { return symbol > rhs.symbol; }
+    inline bool operator>=(const Move& rhs) const { return symbol >= rhs.symbol; }
+
+    StateSet::iterator begin() { return targets.begin(); }
+    StateSet::iterator end() { return targets.end(); }
+
+    StateSet::const_iterator cbegin() const  { return targets.cbegin(); }
+    StateSet::const_iterator cend() const { return targets.cend(); }
 };
 
-/// List of transitions from a certain state. Each element holds transitions with a certain symbol.
-using Moves = Mata::Util::OrdVector<Move>;
-/// Transition relation for an NFA. Each index 'i' to the vector represents a state 'i' in the automaton.
-using TransitionRelation = std::vector<Moves>;
+/**
+ * Post is a data structure representing possible transitions over different symbols.
+ * It is an ordered vector containing possible Moves (i.e., pair of symbol and target states.
+ * Vector is ordered by symbols which are numbers.
+ */
+struct Post : private Util::OrdVector<Move> {
+    using iterator = Util::OrdVector<Move>::iterator;
+    using const_iterator = Util::OrdVector<Move>::const_iterator;
+
+    iterator begin() override { return Util::OrdVector<Move>::begin(); }
+    const_iterator begin() const override { return Util::OrdVector<Move>::begin(); }
+    iterator end() override { return Util::OrdVector<Move>::end(); }
+    const_iterator end() const override { return Util::OrdVector<Move>::end(); }
+
+    const_iterator cbegin() const override { return Util::OrdVector<Move>::cbegin(); }
+    const_iterator cend() const override { return Util::OrdVector<Move>::cend(); }
+
+    Post() = default;
+
+    virtual ~Post() = default;
+
+    const_iterator find(const Move& m) const override { return Util::OrdVector<Move>::find(m);}
+    iterator find(const Move& m) override { return Util::OrdVector<Move>::find(m);}
+
+    void insert(const Move& m) override { Util::OrdVector<Move>::insert(m); }
+
+    Move& operator[](Symbol s)
```

Comment:
One can use const cast, courtesy of Juraj, in order not to duplicate code. https://stackoverflow.com/questions/123758/how-do-i-remove-code-duplication-between-similar-const-and-non-const-member-func
  Or, rather, use find or something like that (use [] only when it mean really accessing an index in a vector, array, ...)?
    Easy to change, the operator is not used anywhere :).